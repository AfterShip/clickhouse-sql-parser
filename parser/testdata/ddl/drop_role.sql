DROP ROLE IF EXISTS r1_01293, r2_01293, r3_01293, r4_01293, r5_01293, r6_01293, r7_01293, r8_01293, r9_01293;
DROP ROLE IF EXISTS r2_01293_renamed;
DROP ROLE IF EXISTS r1_01293@'%', 'r2_01293@%.myhost.com';
