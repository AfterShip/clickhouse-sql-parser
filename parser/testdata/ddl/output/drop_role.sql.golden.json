[
  {
    "DropPos": 0,
    "Target": "ROLE",
    "StatementEnd": 108,
    "Names": [
      {
        "Name": {
          "Name": "r1_01293",
          "QuoteType": 1,
          "NamePos": 20,
          "NameEnd": 28
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r2_01293",
          "QuoteType": 1,
          "NamePos": 30,
          "NameEnd": 38
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r3_01293",
          "QuoteType": 1,
          "NamePos": 40,
          "NameEnd": 48
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r4_01293",
          "QuoteType": 1,
          "NamePos": 50,
          "NameEnd": 58
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r5_01293",
          "QuoteType": 1,
          "NamePos": 60,
          "NameEnd": 68
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r6_01293",
          "QuoteType": 1,
          "NamePos": 70,
          "NameEnd": 78
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r7_01293",
          "QuoteType": 1,
          "NamePos": 80,
          "NameEnd": 88
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r8_01293",
          "QuoteType": 1,
          "NamePos": 90,
          "NameEnd": 98
        },
        "Scope": null,
        "OnCluster": null
      },
      {
        "Name": {
          "Name": "r9_01293",
          "QuoteType": 1,
          "NamePos": 100,
          "NameEnd": 108
        },
        "Scope": null,
        "OnCluster": null
      }
    ],
    "IfExists": true,
    "Modifier": "",
    "From": null
  },
  {
    "DropPos": 110,
    "Target": "ROLE",
    "StatementEnd": 146,
    "Names": [
      {
        "Name": {
          "Name": "r2_01293_renamed",
          "QuoteType": 1,
          "NamePos": 130,
          "NameEnd": 146
        },
        "Scope": null,
        "OnCluster": null
      }
    ],
    "IfExists": true,
    "Modifier": "",
    "From": null
  },
  {
    "DropPos": 148,
    "Target": "ROLE",
    "StatementEnd": 204,
    "Names": [
      {
        "Name": {
          "Name": "r1_01293",
          "QuoteType": 1,
          "NamePos": 168,
          "NameEnd": 176
        },
        "Scope": {
          "LiteralPos": 178,
          "LiteralEnd": 179,
          "Literal": "%"
        },
        "OnCluster": null
      },
      {
        "Name": {
          "LiteralPos": 183,
          "LiteralEnd": 204,
          "Literal": "r2_01293@%.myhost.com"
        },
        "Scope": null,
        "OnCluster": null
      }
    ],
    "IfExists": true,
    "Modifier": "",
    "From": null
  }
]